src/router-view.js

```diff
@@ -37,17 +37,23 @@ const swapStrategies = new SwapStrategies();
 
 @customElement('router-view')
 @noView
-@inject(DOM.Element, Container, ViewSlot, Router, ViewLocator, CompositionTransaction)
+@inject(DOM.Element, Container, ViewSlot, Router, ViewLocator, CompositionTransaction, CompositionEngine)
 export class RouterView {
   @bindable swapOrder;
+  @bindable layout;
+  @bindable layoutView;
```

Comment:
@alvarezmario Yep there are probably a couple of lines that need refactoring/removing as it was a back/forth job with a couple of contributors. There's also bug too which I've found/fixed on my local. Just wanted to get the code out there/some discussion going really.